webapp/graphite/readers.py

```diff
@@ -188,6 +188,86 @@ def fetch(self, startTime, endTime):
     return time_info, values
 
 
+class CarbonCacheReader(object):
+  __slots__ = ('metric')
+  supported = True
+
+  def __init__(self, metric):
+    self.metric = metric
+
+  def get_intervals(self):
+    # intervals doesn't matter in such type of reader
+    # Let's return time.time()
+    start = time.time()
+    end = start
+    return IntervalSet( [Interval(start, end)] )
+
+  def fetch(self, startTime, endTime):
```

Comment:
This method is ~80 LOC. That's a good sign that factoring methods is valuable.

I see three main functions that might be splittable here?

1. formatting and extracting from/until time
2. calculating the step
3. querying and formatting cache data.

If you could refactor this a bit and move that functionality out the methods, this function would be much more readable for the next person :)